### PR TITLE
OVA: match the correct vm for the firmware value

### PR DESCRIPTION
**Pull request by bkhizgiy**
currently for some cases the vm is not being match correctly in the builder which leads to not fetching the firmware value set by virt-v2v, which lead to that the default firmware value being set for bios VMs.

* need to be verified -  the bug is currently randomly reproducing.

### CODE REVIEW

File: pkg/controller/plan/adapter/ova/builder.go [hunk 1]
```diff
@@ -332,15 +332,18 @@ func (r *Builder) mapCPU(vm *model.VM, object *cnv.VirtualMachineSpec) {
...
 	if vm.Firmware == "" {
 		for _, vmConf := range r.Migration.Status.VMs {
-			if vmConf.ID == vm.ID {
+			if vmConf.ID == vmRef.ID {
 				virtV2VFirmware = vmConf.Firmware
 				break
 			}
 		}
+		if virtV2VFirmware == "" {
+			r.Log.Info("failed to match the vm, model VMid is %s, vmRef ID is %s", vm.ID, vmRef.ID)
```
Comment (liranr23) [inline pkg/controller/plan/adapter/ova/builder.go:345]:
```suggestion
			r.Log.Info("failed to match the vm, model vmID is %s, vmRef ID is %s", vm.ID, vmRef.ID)
```
or `vm id`? this type of `VMid` is unusual

**Review by liranr23 — COMMENTED** [review]
i find it very strange that the model vm id differs from the reference.
do you have a clue how or why it happened?

File: pkg/controller/plan/adapter/ova/builder.go (hunk 1 above)
Comment (ahadas) [reply pkg/controller/plan/adapter/ova/builder.go:345]: yep, should be vmID o vmId

Comment (ahadas) [reply pkg/controller/plan/adapter/ova/builder.go:345]: yep, should be vmID o vmId
```suggestion
			r.Log.Info("failed to match the vm", "model vmId", vm.ID, "vmRef ID", vmRef.ID)
```

**Review by ahadas — APPROVED** [review]

**Pull request merged**